supplemental_notebooks/2021-06-29_muddiest_points.Rmd

```diff
@@ -0,0 +1,192 @@
+
+Finally, the normalized counts are computed by dividing the raw counts for each gene by the respective size factor for that cell.
+The normalized counts matrix is then transformed using a log transformation. 
+
+There are a few reasons we typically perform log transformation of our matrix: 
+
+- Log transformation mitigates the relationship between high gene expression and high variance  
+- Reduces general skewness of the data to fit the assumptions of downstream tools 
+
+Prior to log transformation, a pseudocount will also be added to our normalized matrix to avoid undefined values of log(0). 
+The typical pseudocount used is 1, as a pseudocount of 1 preserves sparsity from the original matrix. 
+
+### Comparison of Normalization Methods
+
+Let's go ahead and take a closer look at both of these methods and compare how using the library scaling method versus the pooled scaling method/ deconvolution method affects our dataset. 
+
+Rather than use the original bladder dataset from the Tabula Muris project that we looked at when we first learned normalization, let's take a closer look at the dataset from the Hodgkin's lymphoma data that we filtered and normalized in the `04-dimension_reduction_scRNA.Rmd` notebook.
+In that notebook, we performed normalization using pooled scaling with `scran()` and `scater()` but let's see what would have happened if we hadn't used pooled scaling instead and how that might have affected our downstream analysis. 
+
+```{r setup}
+# for these methods we are using clustering so will need to set our seed first
+set.seed(12345)
+
+# import packages 
+# Magrittr for the pipe %>%
+library(magrittr)
+
+# Packages for single cell processing
+library(scater)
+library(scran)
+```
+
+```{r}
+## set up file paths 
+# main data directory
+data_dir <- file.path("data", "hodgkins")
+
+# Normalized count matrix file 
+normalized_sce_file <- file.path(data_dir, "normalized", "normalized_hodgkins_sce.rds")
+```
+
+```{r}
+# read in normalized hodgkins sce object
+hodgkins_sce <- readr::read_rds(hodgkins_sce_file)
+```
+
+We will go ahead and use the `scater::librarySizeFactors` function to compute the size factors for our counts matrix.
+Then we can use the same `scater::logNormCounts` that we used in class to apply the size factors to the counts matrix, but here we will have to specify the vector containing the size factors we would like to use since they are not in the `colData` yet. 
+
+```{r}
+# calculate library size factors on counts matrix from hodgkins_sce
+library_sf <- scater::librarySizeFactors(counts(hodgkins_sce))
+
+# normalize and log transform counts using library size factors 
+library_sf_sce <- scater::logNormCounts(hodgkins_sce, size.factors = library_sf)
+```
+
+Let's take a look now and compare these two different types of analysis using our PCA analysis. 
+Rather than use `calculatePCA()` though we can use `runPCA()` and add the information directly to our two `SingleCellExperiments`. 
```

Comment:
Since we have now shown this a few times, we can simplify.
```suggestion
We will use `runPCA()` and add the results directly to our two `SingleCellExperiments`. 
```